libs/custom-paper-handler/src/driver/mock_driver.test.ts

```diff
@@ -0,0 +1,203 @@
+import { ImageData, writeImageData } from '@votingworks/image-utils';
+import { mockOf } from '@votingworks/test-utils';
+
+import { PaperHandlerStatus } from './coders';
+import {
+  MockPaperHandlerDriver,
+  MockPaperHandlerStatus,
+  isMockPaperHandler,
+} from './mock_driver';
+import * as statusHelpers from './scanner_status';
+import { MinimalWebUsbDevice } from './minimal_web_usb_device';
+import { PaperHandlerDriver } from './driver';
+
+jest.mock(
+  '@votingworks/image-utils',
+  (): typeof import('@votingworks/image-utils') => ({
+    ...jest.requireActual('@votingworks/image-utils'),
+    writeImageData: jest.fn(),
+  })
+);
+
+function newMockDriver() {
+  const mockDriver = new MockPaperHandlerDriver();
+
+  const operationDelayMs = 1000;
+  mockDriver.setMockOperationDelayMs(operationDelayMs);
```

Comment:
I'm trying to understand the intended usage of this delay. Is the idea that we just need some known delay so we know when to expect states when manipulating Jest fake timers?

Reply:
Starting to rethink whether this is necessary - this was more intended as a way to simulate the real-world delay between sending a command and the paper handler actually physically completing the task. In some cases, the mock was transitioning through states too quickly to be able to see in-between states in the UI, and in particular, I was testing forcing a jam state while a print/scan was in progress. But these feel a little too niche to work into the mock/dev dock, so might leave out for now

Reply:
Yeah, I haven't been very happy with our ability to reliably and exhaustively test states. Our current state machine tests use `xstate`'s `onTransition`, which behaves unreliably in that not all states trigger the `onTransition` handler. It looks like others have had this issue and `onTransition` was removed in v5 in favor of `subscribe`: https://stately.ai/docs/migration#use-subscribe-for-changes-not-ontransition.

That said, since it's way out of scope in the foreseeable future to validate and upgrade to xstate v5, the mock delay approach seemed fine with me. But I'll defer to you on whether that approach would be more trouble to maintain than it's worth.